src/WebJobs.Script.Scaling/AzureEnvironments.cs

```diff
@@ -0,0 +1,200 @@
+                }
+
+                return _homeStampName;
+            }
+
+            set
+            {
+                _homeStampName = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets current worker stamp name
+        /// </summary>
+        public static string CurrentStampName
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_currentStampName))
+                {
+                    _currentStampName = Environment.GetEnvironmentVariable("WEBSITE_CURRENT_STAMPNAME").ToLowerInvariant();
+                }
+
+                return _currentStampName;
+            }
+
+            set
+            {
+                _currentStampName = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets current worker name
+        /// </summary>
+        public static string WorkerName
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_workerName))
+                {
+                    using (var key = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\IIS Extensions\DwasMod"))
+                    {
+                        _workerName = ((string)key.GetValue("IpAddress")).ToLowerInvariant();
+                    }
+                }
+
+                return _workerName;
+            }
+
+            set
+            {
+                _workerName = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets validate SSL certificate
+        /// </summary>
+        public static bool? ValidateCertificates
+        {
+            get
+            {
+                if (_validateCertificates == null)
+                {
+                    using (var key = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\IIS Extensions\DwasMod"))
+                    {
+                        _validateCertificates = (int)key.GetValue("ValidateCertificates") != 0;
+                    }
+                }
+
+                return _validateCertificates;
+            }
+
+            set
+            {
+                _validateCertificates = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets current storage connection string
+        /// </summary>
+        public static string StorageConnectionString
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_storageConnectionString))
+                {
+                    _storageConnectionString = Environment.GetEnvironmentVariable("AzureWebJobsStorage");
+                }
+
+                return _storageConnectionString;
+            }
+
+            set
+            {
+                _storageConnectionString = value;
+            }
+        }
+
+        public static string ManagerPartitionKey
+        {
+            get { return string.Format("{0}(manager)", AzureEnvironments.SiteName); }
```

Comment:
For the worker table, the partition key is (runtime) site name and the row key is stamp/workerIP.   There is a special partition (siteName(manager)) to indicate what is the current manager.